cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -328,3 +329,28 @@ func ValidateAPIEndpoint(c *kubeadm.MasterConfiguration, fldPath *field.Path) fi
 	}
 	return allErrs
 }
+
+// ValidateIgnoreChecksErrors validates duplicates in ignore-checks-errors flag.
+func ValidateIgnoreChecksErrors(ignoreChecksErrors []string, skipPreflightChecks bool) (sets.String, error) {
+	ignoreErrors := sets.NewString()
+	allErrs := field.ErrorList{}
+
+	// TODO: remove once deprecated flag --skip-preflight-checks is removed.
+	if skipPreflightChecks {
+		ignoreErrors.Insert("all")
+	}
+	for _, item := range ignoreChecksErrors {
+		lcItem := strings.ToLower(item) // parameters are case insensitive
+		if ignoreErrors.Has(lcItem) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-checks-errors"), item, "duplicate value"))
+			continue
+		}
+		ignoreErrors.Insert(strings.ToLower(item))
+	}
+
+	if ignoreErrors.Has("all") && ignoreErrors.Len() > 1 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-checks-errors"), strings.Join(ignoreErrors.List(), ","), "don't specify individual checks if 'all' is used"))
```

Comment:
Again, if I say --ignore-checks-errors=Foo,Bar,all then my intention is also unambiguous - I want everything ignored.  The same wrapper-automation use case applies.

Reply:
Same here: "think twice when copy&paste from stackoverflow". but I'm not insisting, if others will see that as too strict.

Reply:
+1 to @kad here